src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -28,48 +37,308 @@ export class TypescriptCodeLensProvider implements vscode.CodeLensProvider {
+
+        // Make function handler relative to baseDir
+        const handlerFileRelativePath = path.relative(
+            rootCodeFolder,
+            path.dirname(this.localInvokeArgs.document.uri.fsPath)
         )
+
+        const relativeFunctionHandler = path.join(
+            handlerFileRelativePath,
+            this.localInvokeArgs.handlerName
+        ).replace('\\', '/')
+
+        await new SamTemplateGenerator()
+            .withCodeUri(rootCodeFolder)
+            .withFunctionHandler(relativeFunctionHandler)
+            .withResourceName(LocalLambdaRunner.TEMPLATE_RESOURCE_NAME)
+            .withRuntime(this.runtime)
+            .generate(inputTemplatePath)
+
+        return inputTemplatePath
     }
 
-    public static async onLambdaLocalInvoke(
-        args: LambdaLocalInvokeArguments,
+    private async determineRootCodeFolder(): Promise<string> {
+        const packageJsonPath: string | undefined =
+            await filesystemUtilities.findFileInParentPaths(
+                path.dirname(this.localInvokeArgs.document.uri.fsPath),
+                'package.json'
+            )
+
+        if (!packageJsonPath) {
+            throw new Error(
+                localize(
+                    'AWS.error.sam.local.package.json.not.found',
+                    'Unable to find package.json related to {0}',
+                    this.localInvokeArgs.document.uri.fsPath
+                )
+            )
+        }
+
+        return path.dirname(packageJsonPath)
+    }
+
+    private async executeSamBuild(
+        rootCodeFolder: string,
+        inputTemplatePath: string
+    ): Promise<string> {
+        this.outputChannel.appendLine(
+            localize(
+                'AWS.output.building.sam.application',
+                'Building SAM Application...'
+            )
+        )
+
+        const samBuildOutputFolder = path.join(await this.getBaseBuildFolder(), 'output')
+
+        await new SamCliBuildCommand(
+            samBuildOutputFolder,
+            rootCodeFolder,
+            inputTemplatePath
+        ).execute()
+
+        this.outputChannel.appendLine(
+            localize(
+                'AWS.output.building.sam.application.complete',
+                'Build complete.'
+            )
+
+        )
+
+        return path.join(samBuildOutputFolder, 'template.yaml')
+    }
+
+    private async invokeLambdaFunction(
+        samTemplatePath: string,
+        debugPort: number | undefined,
     ): Promise<void> {
-        throw new Error('not implemented')
+        this.outputChannel.appendLine(
+            localize(
+                'AWS.output.starting.sam.app.locally',
+                'Starting the SAM Application locally (see Terminal for output)'
+            )
+        )
+
+        let debugPortStr: string | undefined
+
+        if (this.localInvokeArgs.debug) {
+            debugPortStr = debugPort!.toString()
```

Comment:
The user *should* provide a debug port if attempting to debug, but we still need to validate that they did.

Reply:
This is the responsibility of the caller code. The thought is in the future, we might detect a free port and pass that in. However, in this method (`invokeLambdaFunction`), we treat the absence of that arg to mean "run locally, do not debug".  I will add a function header here.